File: checkPalindrome.js
function checkPalindrome(inputString) {
    for(var a = 0; a < inputString.length/2; a++){
       for(var z = inputString.length-1; z > a; z--){
            if(inputString[a] == inputString[z]){
                a++;
            }else{
                return false;
            }
        } 
    }
    return true;
}

File: Java/checkPalindrome.java
class CheckPalindrome {
  boolean checkPalindrome(String str) {
    int mid = str.length()/2;
    String str1 = str.substring(0,mid);
    str1 = new StringBuilder(str1).reverse().toString();
    String str2 = str.length() % 2 == 0 ? str.substring(mid,str.length()) : str.substring(mid+1,str.length());
    
    if(str.length() == 1){
        return true;
    }else {
        return str1.equals(str2);   
    }
  }
}  
